core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -149,13 +149,6 @@ public static long getLong(String property, JsonNode node) {
     return pNode.asLong();
   }
 
-  public static Boolean getBoolOrNull(String property, JsonNode node) {
```

Comment:
I don't feel strongly about this but I think keeping this method is ok, there are tests added as well. I know we may want to keep unused code out though but the other getter's have this option 🤷

Reply:
This caused a revapi failure, so I'll add it back.